src/parser/name_resolution.c

```diff
@@ -10234,3 +10234,80 @@ pt_bind_name_to_spec (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
   node->info.name.meta_class = PT_NORMAL;	// so far, only normals are used.
   return node;
 }
+
+static void
+pt_resolve_dblink_server_name (PARSER_CONTEXT * parser, PT_NODE * node)
+{
+  PT_NODE *val[3];
+  DB_VALUE values[3];
+  PT_DBLINK_INFO *dblink_table = &node->info.dblink_table;
+  int i;
+
+  assert (dblink_table->conn && dblink_table->conn->node_type == PT_NAME);
+
+  db_make_null (&(values[0]));
+  db_make_null (&(values[1]));
+  db_make_null (&(values[2]));
+  i = get_dblink_info_from_dbserver (parser, dblink_table->conn->info.name.original, values);
+  if (i != NO_ERROR)
+    {
+      if (i == ER_DBLINK_SERVER_NOT_FOUND)
+	{
+	  PT_ERRORmf (parser, node, MSGCAT_SET_PARSER_RUNTIME, MSGCAT_RUNTIME_RT_SERVER_NOT_DEFINED,
+		      dblink_table->conn->info.name.original);
+	}
+      else if (i == ER_DBLINK_CANNOT_UPDATE_SERVER)
+	{
+	  PT_ERRORmf (parser, node, MSGCAT_SET_PARSER_RUNTIME, MSGCAT_RUNTIME_RT_SERVER_ALTER_NOT_ALLOWED, 0);
+	}
+      else if (!pt_has_error (parser))
```

Comment:
"if (pt_has_error (parser))" is right?
!pt_has_error means no error occurred.

Reply:
Yes, that's right.
If an error occurs but an error message is not set, you need to set an error message, so this is what you are asking.
In the returned function, it checks the necessity of adding it because it uses pt_has_error (parser) to check the error.

Reply:
If i is not in ('ER_DBLINK_SERVER_NOT_FOUND', 'ER_DBLINK_CANNOT_UPDATE_SERVER') then error is not handled.

Reply:
It is not a problem because the following functions are error-checking using pt_has_error (parser).
For this, I wrote PT_ERROR (parser, node, "Failed to obtain server information");

Reply:
I understand what you said.